rgdps/usecases/user_relationships.py

```diff
@@ -0,0 +1,160 @@
+from __future__ import annotations
+
+from datetime import datetime
+from typing import NamedTuple
+
+from rgdps import repositories
+from rgdps.common.context import Context
+from rgdps.constants.errors import ServiceError
+from rgdps.constants.users import UserRelationshipType
+from rgdps.models.user import User
+from rgdps.models.user_relationship import UserRelationship
+
+
+class UserRelationshipResponse(NamedTuple):
+    relationship: UserRelationship
+    user: User  # It's the user2.
```

Comment:
Might want to make this more clear through a name like `recipient_user` etc. Might be better to use that sort of naming throughout the whole thing? Could also be `target_user`.

Reply:
fixed in 9b37790